SignalServiceKit/src/Loki/Utilities/LokiPushNotificationManager.swift

```diff
@@ -70,4 +73,24 @@ final class LokiPushNotificationManager : NSObject {
             print("[Loki] Couldn't register device token.")
         })
     }
+    
+    @objc(acknowledgeDeliveryForMessageWithHash:expiration:hexEncodedPublicKey:)
+    static func acknowledgeDeliveryForMessage(with hash: String, expiration:Int, hexEncodedPublicKey: String) {
+        let parameters: [String : Any] = [ "lastHash" : hash,
+                                           "pubKey" : hexEncodedPublicKey,
+                                           "expiration": expiration]
+        let url = URL(string: server + "acknowledge_message_delivery")!
+        let request = TSRequest(url: url, method: "POST", parameters: parameters)
+        request.allHTTPHeaderFields = [ "Content-Type" : "application/json" ]
+        TSNetworkManager.shared().makeRequest(request, success: { _, response in
+            guard let json = response as? JSON else {
+                return print("[Loki] Couldn't acknowledge the delivery for message with last hash: " + hash)
+            }
+            guard json["code"] as? Int != 0 else {
```

Comment:
Is this the status code? If so we should probably check that it's 200, not just non-zero

Reply:
This is not the status code... The RESTful API will return a code representing the validation of parameters

Reply:
Okay, all good then